### PR TITLE
Added date-time, range-picker, and time-picker - supports validation

### PR DESCRIPTION
**Pull request by srowhani**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #patch# - backwards-compatible bug fix
 - [ ] #minor# - adding functionality in a backwards-compatible manner
 - [x] #major# - incompatible API change

# CHANGELOG
- `frost-date-time-picker`
- `frost-range-picker`
- `frost-time-picker`
-  validation (wip)
- scss changes

# TODO
- [x] testing
- [ ] coverage
- [x] lint fix
- [x] validation
- [x] datetime
- [x] rangepicker
- [x] timepicker
- [x] update demo

### CODE REVIEW

File: addon/styles/addon.scss [hunk 1]
```diff
@@ -1,24 +1,83 @@
 @import 'node_modules/ember-frost-core/addon/styles/frost-theme';
 @import 'bower_components/pikaday/scss/pikaday';
+@import 'bower_components/clockpicker-seconds/dist/jquery-clockpicker.min';
```
Comment (srowhani) [inline addon/styles/addon.scss]: will have to remove once styling is done

File: addon/templates/components/frost-date-picker.hbs [hunk 2]
```diff
@@ -50,10 +50,12 @@
   escape-press=_eventProxy.escape-press
   insert-newline=_eventProxy.enter
 }}
+{{#unless hideIcon}}
```
Comment (srowhani) [inline addon/templates/components/frost-date-picker.hbs]: should use frost-icon for this

File: tests/dummy/app/pods/date-time/template.hbs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<h2>datetime</h2>
+<p>see logs</p>
```
Comment (rox163) [inline tests/dummy/app/pods/date-time/template.hbs]: can we use notifications instead of logs here?

File: package.json [hunk 4]
```diff
@@ -51,8 +51,10 @@
     "ember-frost-core": "^1.0.0",
     "ember-hook": "^1.3.5",
     "ember-load-initializers": "^0.5.0",
+    "ember-prop-types": "3.2.1",
```
Comment (rox163) [inline package.json]: prop types is already in dependencies

File: addon/components/frost-range-picker.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    onError: PropTypes.func,
+    validator: PropTypes.func
+  },
+
+  /** @returns {Object} the default property values when not provided by consumer */
+  getDefaultProps () {
+    return {
+      startValidator: this.get('isValid'),
+      endValidator: this.get('isValid'),
+      isVertical: false
```
Comment (rox163) [inline addon/components/frost-range-picker.js]: should add hook prop here

File: tests/dummy/app/pods/date-time/controller.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import Ember from 'ember'
+
+export default Ember.Controller.extend({
+  actions: {
+    datetimeChanged (datetime) {
+      console.log(datetime)
```
Comment (rox163) [inline tests/dummy/app/pods/date-time/controller.js]: add a notification indicating a successful selection here.

File: tests/dummy/app/pods/range-picker/template.hbs [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<h2> range picker </h2>
```
Comment (rox163) [inline tests/dummy/app/pods/range-picker/template.hbs]: needs the snippet block

File: tests/dummy/app/pods/range-picker/controller.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import Ember from 'ember'
+
+export default Ember.Controller.extend({
+  actions: {
+    log (e) {
+      // perform validation here
+      console.log(e)
```
Comment (rox163) [inline tests/dummy/app/pods/range-picker/controller.js]: notifications would be nice here. also, example of validation and put it in a snippet

File: tests/dummy/app/pods/range-picker/template.hbs [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+{{frost-range-picker
+  onError=(action 'log')
```
Comment (rox163) [inline tests/dummy/app/pods/range-picker/template.hbs]: should have different functions for each action

File: tests/dummy/app/pods/date-time/template.hbs (hunk 3 above)
Comment (rox163) [inline tests/dummy/app/pods/date-time/template.hbs]: snippet block

File: tests/dummy/app/pods/date-time/controller.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import Ember from 'ember'
+const {
+  inject: {service}
```
Comment (rox163) [inline tests/dummy/app/pods/date-time/controller.js]: fyi, our agreed upon 'ciena' style here is to use up to only one level of destructuring for imports. so just up to the 'inject'.

File: tests/dummy/app/pods/demo/controller.js [hunk 11]
```diff
@@ -2,12 +2,12 @@ import Ember from 'ember'
 
 const {
   Controller,
-  inject
+  inject: {service}
```
Comment (rox163) [inline tests/dummy/app/pods/demo/controller.js]: ditto about destructuring

File: tests/dummy/app/pods/range-picker/controller.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
+import Ember from 'ember'
+
+const {
+  inject: {service}
```
Comment (rox163) [inline tests/dummy/app/pods/range-picker/controller.js]: ditto

**Comment by rox163** [comment]
:+1:

**Comment by rox163** [comment]
Based on UX feedback - add left/right borders as well to make it look like a complete field

**Comment by rox163** [comment]
:+1:

**Pull request merged**